### PR TITLE
[DUOS-2898] Switch more hyperscript-helpers to JSX

### PR DESCRIPTION
**Pull request by fboulnois**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2898

### Summary

Follow up to #2397 and #2438 . Switches more components from `react-hyperscript-helpers` to JSX.

Note: now that we are getting down to the last set of components, some of the code requires functional changes to work. Please test these out.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/SearchSelectOrText.js [hunk 1]
```diff
@@ -68,57 +68,53 @@ export const SearchSelectOrText = (props) => {
...
+    <div className="dropdown select-dropdown" id={id} name={label}>
+      <a
+        className={`btn select-btn btn-secondary dropdown-toggle ${errored ? 'errored' : ''}`}
+        role="button"
+        id="dropdownMenuLink"
+        data-toggle="dropdown"
+        aria-haspopup="true"
+        aria-expanded="false"
+        onClick={onOpen}
+        disabled={disabled}
```
Comment (rushtong) [inline src/components/SearchSelectOrText.js:80]: It looks like this attribute is not allowed here (and indeed, there are other non-standard attributes scattered across the older hyperscript), but I favor keeping this (and others) here to lessen the potential for regressions.

File: src/components/data_update/DatasetUpdate.js [hunk 2]
```diff
@@ -133,261 +132,255 @@ export const DatasetUpdate = (props) => {
...
-        type: 'submit',
-        onClick: submitForm,
-      }, 'Submit'),
-    ]),
-  ]);
+  return (
+    <div className='data-update-section'>
+      <h2>1. Dataset Information</h2>
+      <FormField
+        id="datasetName"
```
Comment (rushtong) [inline src/components/data_update/DatasetUpdate.js:139]: I'm curious about the double quotes not triggering our eslint rule:
```
"quotes": ["warn", "single", {"avoidEscape": true, "allowTemplateLiterals": true}], // Checks for single quote usage where possible
```
Not critical for this PR, but we should investigate.

**Review by rushtong — APPROVED** [review]
Thank you! 👍🏽

**Review by aarohinadkarni — APPROVED** [review]
👍

**Pull request merged**